src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
-                                                   OutletNodeErrFlag);
-                        thisDOAS.m_FanInletNodeNum = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).InletNodeNum(CompNum);
-                        thisDOAS.m_FanOutletNodeNum = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).OutletNodeNum(CompNum);
+                        thisInletNodeNum = Fans::GetFanInletNode(state, typeNameUC, CompName, InletNodeErrFlag);
+                        thisOutletNodeNum = Fans::GetFanOutletNode(state, typeNameUC, CompName, OutletNodeErrFlag);
+                        thisDOAS.m_FanInletNodeNum = thisInletNodeNum;
+                        thisDOAS.m_FanOutletNodeNum = thisOutletNodeNum;
                         if (!(CompNum == 1 || CompNum == state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).NumComponents)) {
                             ShowSevereError(state,
-                                            "The fan placement is either first as blow through or last as draw through in" + CurrentModuleObject +
-                                                " = " + state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum));
+                                            format("The fan placement is either first as blow through or last as draw through in{} = {}",
+                                                   CurrentModuleObject,
+                                                   CompName));
                             ShowContinueError(state, format("The current position is number {}", CompNum));
                             errorsFound = true;
                         }
-                    } else if (SELECT_CASE_var == "COIL:COOLING:WATER") {
-                        state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).InletNodeNum(CompNum) =
-                            WaterCoils::GetCoilInletNode(state,
-                                                         SELECT_CASE_var,
-                                                         state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum),
-                                                         InletNodeErrFlag);
-                        state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).OutletNodeNum(CompNum) =
-                            WaterCoils::GetCoilOutletNode(state,
-                                                          SELECT_CASE_var,
-                                                          state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum),
-                                                          OutletNodeErrFlag);
+                        break;
+
+                    case ValidEquipListType::CoilCoolingWater:
+                        thisInletNodeNum = WaterCoils::GetCoilInletNode(state, typeNameUC, CompName, InletNodeErrFlag);
```

Comment:
Even returning a `std::pair` would be more efficient than calling what is basically the same function twice.

Reply:
That's some next level C++ 💩 !